envoy/stream_info/filter_state.h

```diff
@@ -26,7 +26,24 @@ using FilterStateSharedPtr = std::shared_ptr<FilterState>;
  */
 class FilterState {
 public:
-  enum class StateType { ReadOnly, Mutable };
+  enum StateType {
```

Comment:
I think it's simpler to just leave this as the previous enum, and add another bool for `SharedWithUpstreamConnection`.

Reply:
When the upstream L7 filters get done, we'll also need to share with the upstream request I think. I could add another enum for sharing configuration I suppose?